File: SDLC/SDLC1/W4/Eindopdr/MastermindCore6/MastermindCore6/MastermindCore6/Model/MM.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MastermindCore6.Model
{
    public class MM
    {
        public string TeRaden { get; set; }
        public int Beurt { get; set; }
        public string Poging { get; set; }
        private Random r = new Random();

        // NieuwSpel
        /// <summary>
        /// Start nieuw spel, Beurt wordt 0. TeRaden wordt nieuw getal.
        /// </summary>
        /// <returns>
        /// void
        /// </returns>
        public void NieuwSpel()
        {
            Beurt = 0;
            TeRaden = "";
            char[] mogelijkheden = { '1', '2', '3', '4', '5', '6', '7', '8', '9', '0' };
            char teradencijfer;
            for (int i = 0; i < 4; i++)
            {
                do
                {
                    int x = r.Next(10);
                    teradencijfer = mogelijkheden[x];
                    mogelijkheden[x] = 'x';
                } while (teradencijfer == 'x');
                TeRaden += teradencijfer;
            }
        }

        // NieuweBeurt
        /// <summary>
        /// Beurt wordt opgehoogd
        /// </summary>
        /// <returns>
        /// void
        /// </returns>
        public void NieuweBeurt()
        {
            Beurt++;
        }

        // NieuwePoging
        /// <summary>
        /// Neemt ingevoerde poging over, corrigeert aantal cijfers in poging.
        /// </summary>
        /// <param name="p">string, bevat de ingevoerde poging</param>
        public void NieuwePoging(string p)
        {
            while (p.Length < 4) p = "0" + p;
            if (p.Length > 4) p = p.Remove(4);
            Poging = p;
        }
        // PositiesOK
        /// <summary>
        /// Bereken aantal correcte posities van poging in TeRaden.
        /// </summary>
        /// <returns>
        /// integer, aantal cijfers op de juiste positie
        /// </returns>

        public int PositiesOK()
        {
            int pos = 0;
            for (int i = 0; i < 4; i++) if (TeRaden[i] == Poging[i]) pos++;
            return pos;
        }

        // AantalOK
        /// <summary>
        /// Bereken aantal correcte cijfers van Poging in TeRaden
        /// </summary>
        /// <returns>
        /// integer, aantal correcte cijfers
        /// </returns>
        public int AantalOK()
        {
            int aan = 0;
            string x = TeRaden;
            for (int i = 0; i < 4; i++)
            {
                int y = x.IndexOf(Poging[i]);
                if (y >= 0)
                {
                    aan++;
                    x = x.Remove(y, 1);
                }
            }
            return aan;
        }
    }
}

File: SDLC/SDLC1/W5/mastermindv81/mastermindv8/mastermindv8/app/Models/MM.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

/**
    @class MM
    * @property string TeRaden te raden getal bestaande uit 4 cijfers
    * @property integer Beurt beurtnummer
    * @property array Pogingen bevat de pogingen
    * @property bool OpenSpel keuze of te raden getal zichtbaar is (ontwikkelmodus)
*/

class MM extends Model
{
    public $TeRaden;
    public $Beurt;
    public $Pogingen = array();
    public $OpenSpel = FALSE;

    /**
        * @method NieuwSpel Start nieuw spel.
        * @property Beurt krijgt de waarde 0
        * @property TeRaden krijgt nieuwe waarde
        * @property Pogingen wordt geleegd
        *
        * @param  void
        * @return void
     */
    public function NieuwSpel()
    {
        $this->Beurt = 0;

        $this->TeRaden = "";
        $mogelijkheden = "1234567890";          // deze cijfers mag TeRaden bevatten
        for ($i=0; $i<4; $i++){                 // TeRaden bestaat uit 4 cijfers
            do {
                $x = rand(0,9);
                $teradencijfer = $mogelijkheden[$x];
                $mogelijkheden[$x] = 'x';
                
            } while ($teradencijfer == 'x');    // voorkom dubbelingen in TeRaden
            $this->TeRaden .= $teradencijfer;
        }        
        $this->Pogingen = array();
    }

    /**
        * @method NieuweBeurt Start nieuwe beurt
        * @property Beurt wordt opgehoogd
        * @param  void
        * @return void
     */    
    public function NieuweBeurt()
    {
        $this->Beurt++;
    }
    /**
        * @method PositiesOK Bereken aantal correcte posities van poging in TeRaden.
        * @param string $poging 4 cijfers
        * @return integer aantal correcte posities
     */    

    public function PositiesOK($poging)
    {
        $positiesOK = 0;
        for ($i = 0; $i < 4; $i++) if ( $this->TeRaden[$i] ==  $poging[$i]) $positiesOK++;
        return $positiesOK;
    }
    /**
        * @method AantalOK Bereken aantal correcte cijfers van poging in TeRaden.
        * @param string $poging 4 cijfers
        * @return integer aantal correcte cijfers
     */   

    public function AantalOK($poging)
    {
        $aantalOK = 0;
        $teraden = $this->TeRaden;
        for ($i = 0; $i < 4; $i++)
        {          
            $y = strpos($teraden, $poging[$i]);
            if ($y !== FALSE)
            {
                $aantalOK++;
                $teraden[$y] = 'x';
            }
        }
        return $aantalOK;
    }
}
